### PR TITLE
systemd 242

### PR DESCRIPTION
**Pull request by maxim-belkin**
1. Updated `systemd` version to 242
2. Ran `brew audit --fix`

### CODE REVIEW

**Comment by iMichka** [comment]
I think @sjackman wanted to stay on that old version, because we update systemd only when the OS requirements are bumped.

**Comment by maxim-belkin** [comment]
> because we update systemd only when the OS requirements are bumped.

OK. I can create `system@242`

**Comment by sjackman** [comment]
> I think @sjackman wanted to stay on that old version, because we update systemd only when the OS requirements are bumped.

I don't recall that being the case, but it's been a long time since I've looked at `systemd`. I have no problem with bumping it if it passes tests. It may require also bumping anything that depends on it.

**Comment by maxim-belkin** [comment]
Just in case, the formula is ready https://github.com/maxim-belkin/linuxbrew-core/blob/systemd-242/Formula/systemd.rb

I can submit a PR to update `systemd` or as a new (`systemd@242`) formula.

There is the following issue, however:
1. Technically, formula builds fine without setting `rootprefix` (as in `"-Drootprefix=#{prefix}"` [here](https://github.com/maxim-belkin/linuxbrew-core/blob/systemd-242/Formula/systemd.rb#L39)). If I do that, everything build just fine and the structure of created directory is fine too. **However, none of the binaries can find `libsystemd-shared-242.so`**.
2. `patchelf --print-rpath` on a few of these non-working executables returned `/lib/systemd`.
3. `lib/systemd` is paired with `rootprefixdir` to create `rootlibexecdir`in `meson.build` file.
4. From the same `meson.build` file we can see that `rootprefixdir` is created from `rootprefix` variable. 
5. When I set `rootprefix` directory, the script creates `/home/linuxbrew/.linuxbrew/Cellar/systemd/242` directory **under** `/home/linuxbrew/.linuxbrew/Cellar/systemd/242` and puts produced binaries/libraries/etc in there. So, all the files are placed in
```
/home/linuxbrew/.linuxbrew/Cellar/systemd/242/home/linuxbrew/.linuxbrew/Cellar/systemd/242/
```
As a result, I need to move files from that strange location back to `prefix` -- this is what I do with `cp_r Dir["#{prefix}/#{prefix}/*"], prefix` and `rm_rf Dir["#{prefix}/home"]` lines.

We can't replace `cp_r` + `rm_rf` with `mv` because `mv` overwrites the directories that already exist in `#{prefix}`. Here is what is deleted when `mv` is used instead of `cp_r` + `rm_rf`

```
Only in systemd/242/etc: binfmt.d
Only in systemd/242/etc: kernel
Only in systemd/242/etc: sysctl.d
Only in systemd/242/etc/systemd: network
Only in systemd/242/etc/systemd: system
Only in systemd/242/etc/systemd: user
Only in systemd/242/etc: tmpfiles.d
Only in systemd/242/etc/udev: hwdb.d
Only in systemd/242/etc/udev: rules.d
Only in systemd/242/etc/xdg/systemd: user
Only in systemd/242/lib/systemd/system: runlevel1.target.wants
Only in systemd/242/lib/systemd/system: runlevel2.target.wants
Only in systemd/242/lib/systemd/system: runlevel3.target.wants
Only in systemd/242/lib/systemd/system: runlevel4.target.wants
Only in systemd/242/lib/systemd/system: runlevel5.target.wants
Only in systemd/242/lib/systemd: system-shutdown
Only in systemd/242/lib/systemd: system-sleep
Only in systemd/242/sbin: resolvconf
Only in systemd/242/usr/lib: binfmt.d
Only in systemd/242/usr/lib/environment.d: 99-environment.conf
Only in systemd/242/usr/lib/systemd: user-generators
Only in systemd/242/var: lib
Only in systemd/242/var/log: journal
```

**Comment by maxim-belkin** [comment]
There are unwanted system libraries in my local Docker linuxbrew/linuxbrew-based image (yes, not linuxbrew/brew):

```
$ brew linkage systemd
System libraries:
  /lib/x86_64-linux-gnu/libblkid.so.1
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libcap.so.2
  /lib/x86_64-linux-gnu/libcrypt.so.1
  /lib/x86_64-linux-gnu/libgcrypt.so.20
  /lib/x86_64-linux-gnu/libgpg-error.so.0
  /lib/x86_64-linux-gnu/liblzma.so.5
  /lib/x86_64-linux-gnu/libm.so.6
  /lib/x86_64-linux-gnu/libmount.so.1
  /lib/x86_64-linux-gnu/libpthread.so.0
  /lib/x86_64-linux-gnu/librt.so.1
  /lib64/ld-linux-x86-64.so.2
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/Cellar/systemd/242/lib/systemd/libsystemd-shared-242.so (systemd)
Dependencies with no linkage:
  libcap
Unwanted system libraries:
  /lib/x86_64-linux-gnu/libblkid.so.1
  /lib/x86_64-linux-gnu/libcap.so.2
  /lib/x86_64-linux-gnu/libgcrypt.so.20
  /lib/x86_64-linux-gnu/libgpg-error.so.0
  /lib/x86_64-linux-gnu/liblzma.so.5
  /lib/x86_64-linux-gnu/libmount.so.1
```
But I think I'll wait until the decision is made whether we need to update this formula at all

**Comment by sjackman** [comment]
> But I think I'll wait until the decision is made whether we need to update this formula at all

The path of least resistance is to leave it as is. If someone wants to update it, go ahead, and the usual tests need to pass, including `brew linkage` and `brew test` of all recursive dependencies.

**Comment by maxim-belkin** [comment]
Given that I have already ported configure/make to meson/ninja, I wouldn't call it "least resistance"... 🙃

File: Formula/systemd.rb [hunk 1]
```diff
@@ -18,22 +17,23 @@ class Systemd < Formula
...
-  depends_on "libgpg-error" => :build
+
+  fails_with :gcc => "4.8"
 
   patch do
     url "https://github.com/systemd/systemd/commit/227b8a762fea1458547be2cdf0e6e4aac0079730.diff?full_index=1"
     sha256 "3a123a6cba8cf1e27b2ace04f81c6d0a87e29c7a9900aa2ec1d1fea8e06656a8"
   end
 
   # src/core/dbus.c:1022:5: internal compiler error: Segmentation fault
```
Comment (sjackman) [inline Formula/systemd.rb]: Move this comment along with the `fails_with` line.

Comment (maxim-belkin) [reply Formula/systemd.rb]: OK, will do.

I think I'll open a separate issue to document/explain steps required to update systemd 234 to 242.

**Pull request closed without merging**